lib/programmemory.cpp

```diff
@@ -1270,9 +1270,10 @@ namespace {
                 ValueFlow::Value r = state.execute(tok);
                 if (r.isUninitValue())
                     continue;
-                result.insert(std::make_pair(tok->exprId(), r));
```

Comment:
I filed https://github.com/llvm/llvm-project/issues/79213 about detecting the unnecessary temporary object.